tcp/server.go

```diff
@@ -26,10 +26,10 @@ func NewServer(host string, port int) *Server {
 }
 
 // Listen listens for incoming connections
-func (s *Server) Listen(c Callback) error {
```

Comment:
we still wanna return an error here, for if the server fails to start, that might need to be handled outside of this class